### PR TITLE
Pi-hole

### PR DESCRIPTION
**Pull request by Fredrik81**
Pi-Hole rockon.
This rockon is a DNS forwarder with filtering function.
Website: https://pi-hole.net

This rockon depends on port 80 to work as intended!

### CODE REVIEW

File: pi-hole.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                    "80": {
+                        "description": "Web-port. Recommended: 80",
+                        "host_default": 80,
+                        "label": "Web-Port",
+                        "ui": true
+                    }
+                },
+                "volumes": {
+                    "/etc/pihole": {
+                        "description": "Choose a Share for Plex Requests configuration. Eg: create a Share called pihole-config for this purpose alone.",
```
Comment (florianduclot) [inline pi-hole.json]: Replace old reference to Plex Requests with something related to Pi-hole, such as "Choose a share for Pi-Hole configuration files", for instance.

File: pi-hole.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                        "label": "Rockstor IP",
+                        "index": 1
+                    },
+                    "VIRTUAL_HOST": {
+                        "description": "Enter IP of rockstor server.",
+                        "label": "Virtual Host",
+                        "index": 2
+                    },
+                    "DNS1": {
+                        "description": "Enter desired webpassword for pi-hole.",
```
Comment (florianduclot) [inline pi-hole.json]: Primary upstream DNS for Pi-Hole's DNSMasq to use, defaults to google 8.8.8.8

File: pi-hole.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        "label": "DNS Server 1",
+                        "index": 3,
+                        "default": "8.8.8.8"
+                    },
+                    "DNS2": {
+                        "description": "Enter desired webpassword for pi-hole.",
```
Comment (florianduclot) [inline pi-hole.json]: Secondary upstream DNS for Pi-Hole's DNSMasq to use, defaults to google 8.8.4.4

File: pi-hole.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        "label": "DNS Server 2",
+                        "index": 4,
+                        "default": "8.8.4.4"
+                    },
+                    "IPv6": {
+                        "description": "Enter desired webpassword for pi-hole.",
+                        "label": "Enable IPv6",
+                        "index": 5,
+                        "default": "false"
```
Comment (florianduclot) [inline pi-hole.json]: The wording of the container's documentation for this option is a little ambiguous to me, but it seems that if set as "True", ads using IPv6 addresses will be blocked. Then, if set to "False", these ads will not be blocked. If my reasoning is correct, shouldn't this option be left as "false"?

**Comment by Fredrik81** [comment]
Thanks for the comments. I made changes to the file.

File: pi-hole.json (hunk 4 above)
Comment (coleberhorst) [inline pi-hole.json]: I have it as false on my setup and the web ui allows you to change this setting so probably best left default off.

File: pi-hole.json [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                	"always"
+            	    ]
+                ],
+                "launch_order": 1,
+                "ports": {
+                    "53": {
+                        "description": "DNS port. Required: 53",
+                        "host_default": 53,
+                        "label": "DNS-Port",
+                        "ui": false
```
Comment (coleberhorst) [inline pi-hole.json]: Remove  "ui": false
and the comma from  "label": "DNS-Port",

File: pi-hole.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                        "label": "Web-Password",
+                        "index": 6
+                    }
+                }
+            }
+        },
+        "description": "PI-Hole by DigInc",
+        "more_info": "<h4>PI-HOLE™: A BLACK HOLE FOR INTERNET ADVERTISEMENTS</h4><p><b>Admin page</b></p><p>To access admin interface go to URL: http://[SERVERIP]/Admin</p><p>If you have different port than 80 you need to specify that in the URL.</p><p><b>Block Over 100,000 Ad-serving Domains</b></p><p>Known ad-serving domains are pulled from third party sources and compiled into one list.</p><p><b>Block Advertisements On Any Device</b></p><p>Network-level blocking allows any device to block ads, regardless of hardware or OS.</p><p><b>Improve Overall Network Performance</b></p><p>Since ads are blocked before they are downloaded, your network will perform better.</p>",
+        "ui": {
+            "slug": "admin"
```
Comment (coleberhorst) [inline pi-hole.json]: add a trailing slash to admin/

 "slug": "admin/"

Comment (coleberhorst) [inline pi-hole.json]: File should look like this afterward except the json extension not the txt extension.
[pi-hole.txt](https://github.com/rockstor/rockon-registry/files/1016514/pi-hole.txt)

**Comment by magicalyak** [comment]
Do you think this is ready?

**Comment by coleberhorst** [comment]
not quite until the changes to the web ui link are made. It does work though.

**Comment by Fredrik81** [comment]
@coleberhorst can you elaborate what is missing? I tested it with the latest changes you contributed with and it's all working for me.

**Comment by coleberhorst** [comment]
i apologize i didn't look before i posted :) looks good, does the admin link work from the rock-ons page? I'd say it is ready for merge if it does.

**Comment by magicalyak** [comment]
Great. I'll run a test and review soon.

**Comment by magicalyak** [comment]
I need to run a secondary system to test, my primary runs named already so pretty sure this would interfere.  I'll try to test soon though.

**Comment by phillxnet** [comment]
@magicalyak just linking to @Fredrik81 's forum post introducing this pull request where the rest of the thread provides some context to the development / debugging / testing of this Rock-on.

**Comment by Fredrik81** [comment]
@phillxnet All the "bugs" have been corrected. So the rock-on should be ready to be merged.

**Comment by schakrava** [comment]
Thanks @Fredrik81 @coleberhorst @florianduclot @phillxnet and @magicalyak

**Pull request merged**